packages/app/src/components/DBRowSidePanelHeader.tsx

```diff
@@ -97,16 +139,40 @@ export default function DBRowSidePanelHeader({
           p="xs"
           mt="sm"
           style={{
-            maxHeight: 120,
+            maxHeight: expandSidebarHeader ? maxBoxHeight : undefined,
             overflow: 'auto',
             overflowWrap: 'break-word',
           }}
+          ref={headerRef}
         >
-          <Flex align="baseline" gap={2} mb="xs">
-            <Text size="xs" c="gray.4">
-              {mainContentHeader}
-            </Text>
-            <EditButton sourceId={sourceId} />
+          <Flex justify="space-between" mb="xs">
+            <Flex align="baseline" gap={2}>
+              <Text size="xs" c="gray.4">
+                {mainContentHeader}
+              </Text>
+              <EditButton sourceId={sourceId} />
+            </Flex>
+            {/* Toggles expanded sidebar header*/}
+            {headerHeight >= maxBoxHeight && (
+              <Button
+                size="compact-xs"
+                variant="subtle"
+                color="gray.3"
+                onClick={() =>
+                  setUserPreference({
+                    ...userPreferences,
+                    expandSidebarHeader: !expandSidebarHeader,
+                  })
+                }
+              >
+                {/* TODO: Only show expand button when maxHeight = 120? */}
+                {expandSidebarHeader ? (
+                  <i className="bi bi-arrows-angle-expand" />
+                ) : (
+                  <i className="bi bi-arrows-angle-contract" />
```

Comment:
wait aren't these backwards? Am I reading this correctly?

if `expandSidebarHeader` is true wouldn't you show the contract icon?